src/plugins/intel_gpu/src/graph/impls/ocl/dft.cpp

```diff
@@ -26,9 +26,33 @@ struct dft_impl : typed_primitive_impl_ocl<dft> {
         auto params = get_default_params<kernel_selector::dft_params>(impl_param);
         auto primitive = arg.get_primitive();
         params.axes = primitive->axes;
+
+        if (primitive->signal_size.empty()) {
+            params.signal_size = std::vector<int64_t>(params.axes.size(), -1);
+        } else {
+            params.signal_size = primitive->signal_size;
+        }
+
         if (primitive->kind == dft_kind::inverse) {
-            params.kind = kernel_selector::dft_params::inverse;
+            params.kind = kernel_selector::dft_params::Kind::inverse;
         }
+        if (primitive->mode == dft_mode::real) {
+            params.mode = kernel_selector::dft_params::Mode::real;
+        }
+
+        // Extend input layout for RDFT case to make input rank match output rank
+        if (primitive->kind == dft_kind::forward && primitive->mode == dft_mode::real) {
+            const auto input_layout = arg.input().get_output_layout();
+            const auto output_layout = arg.get_output_layout();
```

Comment:
Please rebase on top of latest master and adjust this function to use `impl_param` instead of `arg` in create function